docs/recipes.md

```diff
@@ -0,0 +1,309 @@
+array
+```
+
+3. Get the value of the UI elements using `array.value`
+
+```python
+array.value
+```
+
+### Create a dictionary of UI elements 
+
+**Task.** Create a dictionary of UI elements that reacts to user interactions,
+especially when the number of elements is not known until runtime.
+
+**Recipe.**
+
+1. Import packages.
+
+```python
+import marimo as mo
+```
+
+2. Use [`mo.ui.dictionary`](api/inputs/dictionary.md#marimo.ui.dictionary) to
+   group together many UI elements into a list.
+
+```python
+dictionary = mo.ui.dictionary({str(i): mo.ui.text() for i in range(10)})
+dictionary
+```
+
+3. Get the value of the UI elements using `dictionary.value`
+
+```python
+dictionary.value
+```
+
+### Create a form with multiple UI elements
+
+**Task.** Batch multiple UI elements into a submittable form.
+
+**Recipe.**
+
+1. Import packages.
+
+```python
+import marimo as mo
+```
+
+2. Use [`mo.ui.form`](api/inputs/form.md#marimo.ui.form) and
+[`Html.batch`](api/Html.md#marimo.Html.batch) to create a form with
+multiple elements.
+
+```python
+form = mo.md(
+   r"""
+   Choose your algorithm parameters:
+
+   - $\lambda$: {lmbda}
+   - $\rho$: {rho}
+   """
+).batch(lmbda=mo.ui.slider(0.1, 1, step=0.1), rho=mo.ui.number(1, 10)).form()
+form
+```
+
+3. Get the submitted form value.
+
+```python
+form.value
+```
+
+
+## Working with buttons
+
+### Create a counter button
+
+**Task.** Create a button that counts the number of times it's clicked.
+
+**Recipe.**
+
+1. Import packages
+
+```python
+import marimo as mo
+```
+
+2. Use [`mo.ui.button`](api/inputs/button.md#marimo.ui.button) and its
+   `on_click` argument to create a counter button.
+
+```python
+# Initialize the button value to 0, increment it on every click
+button = mo.ui.button(value=0, on_click=lambda count: count + 1)
+button
+```
+
+3. Get the button value
+
+```python
+button.value
+```
+
+### Create a toggle button
+
+**Task.** Create a button that toggles between `True` and `False`. (Tip: you
+can also just use [`mo.ui.switch`](api/inputs/switch.md#marimo.ui.switch).)
+
+**Recipe.**
+
+1. Import packages
+
+```python
+import marimo as mo
+```
+
+2. Use [`mo.ui.button`](api/inputs/button.md#marimo.ui.button) and its
+   `on_click` argument to create a toggle button.
+
+```python
+# Initialize the button value to False, flip its value on every click.
+button = mo.ui.button(value=False, on_click=lambda value: not value)
+button
+```
+
+3. Get the button value.
+
+```python
+button.value
+```
+
+### Reveal an output when a button is pressed
+
+**Task.** Reveal an output when a button is pressed.
+
+**Recipe.**
+
+1. Import packages
+
+```python
+import marimo as mo
+```
+
+2. Create a counter button.
+
+```python
+button = mo.ui.button(value=0, on_click=lambda count: count + 1)
+button
+```
+
+3. Show an output after the button is clicked.
+
+```python
+mo.md("#" + "🍃" * button.value) if button.value > 0 else None
+```
+
+### Run a cell when a buton is pressed
```

Comment:
These button examples are great, could be nice to include a couple one-liners for use cases